### ISSUE TITLE
Allow only one way for importing Gherkin file into tests

### ISSUE THREAD

**Issue by hauleth** [cabbage-ex/cabbage#77]
~~IMHO such case should be disallowed, and for defining "helper modules" there should be separate module that would define `def*` macros.~~

**Comment by hauleth** [comment]
Ok, I found that it makes sense. But alternatively I would suggest using only one way to import Gherkin file.

**Comment by revati** [comment]
Yes it is walid case to have cabbage feature without specified feature file.

But are there multiple ways to import gherkin files? 🤔

**Comment by hauleth** [comment]
Ok, I have read the code wrongly. But anyway, I would disallow such case and instead force user to always define `:file` in use, and instead provide separate module with DSL macros.

**Comment by revati** [comment]
Something like `Cabbage.Feature` and `Cabbage.AbstractFeature`?

While I see a benefit to it, it is more like nice to have feature, not must have, it as well might be a 1.1 release, with an extended feature. 

Currently, my priority is getting [1.0](https://github.com/cabbage-ex/cabbage/projects/2) out.

And I have too few hours in week I can spend on it :(

**Comment by hauleth** [comment]
Quick and dirty example how it could work [`Watermelon.Case`](https://github.com/hauleth/watermelon/blob/55bacb4505765e4065037a2731ac9c4786b6f6a5/lib/watermelon/case.ex) and complementary [`Watermelon.DSL`](https://github.com/hauleth/watermelon/blob/55bacb4505765e4065037a2731ac9c4786b6f6a5/lib/watermelon/dsl.ex). [Example usage of "external step definitions"](https://github.com/hauleth/watermelon/blob/55bacb4505765e4065037a2731ac9c4786b6f6a5/test/watermelon/case_test.exs).
